uniffi_bindgen/src/interface/object.rs

```diff
@@ -413,4 +437,82 @@ mod test {
 
         Ok(())
     }
+
+    #[test]
+    fn test_alternate_constructors() -> Result<()> {
+        const UDL: &str = r#"
+            namespace test{};
+            interface Testing {
+                constructor();
+                [Name=new_with_u32]
+                constructor(u32 v);
+            };
+        "#;
+        let ci = ComponentInterface::from_webidl(UDL).unwrap();
+        assert_eq!(ci.iter_object_definitions().len(), 1);
+
+        let obj = ci.get_object_definition("Testing").unwrap();
+        assert!(obj.primary_constructor().is_some());
+        assert_eq!(obj.alternate_constructors().len(), 1);
+        assert_eq!(obj.methods().len(), 0);
+
+        let cons = obj.primary_constructor().unwrap();
+        assert_eq!(cons.name(), "new");
+        assert_eq!(cons.arguments.len(), 0);
+        assert_eq!(cons.ffi_func.arguments.len(), 0);
+
+        let cons = obj.alternate_constructors()[0];
+        assert_eq!(cons.name(), "new_with_u32");
+        assert_eq!(cons.arguments.len(), 1);
+        assert_eq!(cons.ffi_func.arguments.len(), 1);
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_the_name_new_identifies_the_primary_constructor() -> Result<()> {
+        const UDL: &str = r#"
+            namespace test{};
+            interface Testing {
+                [Name=newish]
+                constructor();
+                [Name=new]
+                constructor(u32 v);
+            };
+        "#;
+        let ci = ComponentInterface::from_webidl(UDL).unwrap();
+        assert_eq!(ci.iter_object_definitions().len(), 1);
+
+        let obj = ci.get_object_definition("Testing").unwrap();
+        assert!(obj.primary_constructor().is_some());
+        assert_eq!(obj.alternate_constructors().len(), 1);
+        assert_eq!(obj.methods().len(), 0);
+
+        let cons = obj.primary_constructor().unwrap();
+        assert_eq!(cons.name(), "new");
+        assert_eq!(cons.arguments.len(), 1);
+
+        let cons = obj.alternate_constructors()[0];
+        assert_eq!(cons.name(), "newish");
+        assert_eq!(cons.arguments.len(), 0);
+        assert_eq!(cons.ffi_func.arguments.len(), 0);
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_the_name_new_is_reserved_for_constructors() {
```

Comment:
Whilst we're crossing the 👀 and dotting the 🍵 testing-wise, what happens if we have two constructors called `new`?

```
            namespace test{};
            interface Testing {
                constructor();
                [Name=new]
                constructor();
            };
```

Reply:
OK, I'll fess up...I actually started writing this exact test and then deleted it when I realized that it was likely to behave in a dumb manner (allowing duplicate names in the `ComponentInterface` and then giving inscrutable compile errors later on in the generated code). I'll take a look at making this error out more cleanly.

Reply:
(Surprise! Making it error out on duplicate names was actually quite straightforward and I should have just done this work the first time around 😅. Updated.)